src/core/core.controller.js

```diff
@@ -594,13 +594,14 @@ class Chart {
 	 */
 	updateDatasets(mode) {
 		const me = this;
+		const isFun = typeof mode === 'function';
 
 		if (plugins.notify(me, 'beforeDatasetsUpdate') === false) {
 			return;
 		}
 
 		for (let i = 0, ilen = me.data.datasets.length; i < ilen; ++i) {
-			me.updateDataset(i, mode);
+			me.updateDataset(i, isFun ? mode(i) : mode);
```

Comment:
should we make this a context object instead of just passing `i` directly?

also, I might call it `isFunc` or `isFunction` or something since "fun" is actually a word

Reply:
done